aiida/engine/processes/exit_code.py

```diff
@@ -8,49 +8,66 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """A namedtuple and namespace for ExitCodes that can be used to exit from Processes."""
-
-from collections import namedtuple
-
 from aiida.common.extendeddicts import AttributeDict
 
 __all__ = ('ExitCode', 'ExitCodesNamespace')
 
-ExitCode = namedtuple('ExitCode', ['status', 'message', 'invalidates_cache'])
-ExitCode.__new__.__defaults__ = (0, None, False)
-"""
-A namedtuple to define an exit code for a :class:`~aiida.engine.processes.process.Process`.
 
-When this namedtuple is returned from a Process._run() call, it will be interpreted that the Process
-should be terminated and that the exit status and message of the namedtuple should be set to the
-corresponding attributes of the node.
+class ExitCode:
+    """A simple data class to define an exit code for a :class:`~aiida.engine.processes.process.Process`.
 
-:param status: positive integer exit status, where a non-zero value indicated the process failed, default is `0`
-:type status: int
+    When an instance of this clas is returned from a `Process._run()` call, it will be interpreted that the `Process`
+    should be terminated and that the exit status and message of the namedtuple should be set to the corresponding
+    attributes of the node.
 
-:param message: optional message with more details about the failure mode
-:type message: str
+    :param status: positive integer exit status, where a non-zero value indicated the process failed, default is `0`
+    :type status: int
 
-:param invalidates_cache: optional flag, indicating that a process should not be used in caching
-:type invalidates_cache: bool
-"""
+    :param message: optional message with more details about the failure mode
+    :type message: str
+
+    :param invalidates_cache: optional flag, indicating that a process should not be used in caching
+    :type invalidates_cache: bool
+    """
+
+    def __init__(self, status=0, message=None, invalidates_cache=False):
+        self.status = status
+        self.message = message
+        self.invalidates_cache = invalidates_cache
+
+    def __call__(self, **kwargs):
```

Comment:
Also not completely sure about using `__call__` for this - if we want to extend the functionality of the `ExitCode` class in the future, it would be slightly arbitrary to give `__call__` to this method. Maybe `evaluate` or `format`?

Reply:
I think the latter conveys the intent a bit better - makes it clear that the parameters are there only for formatting the message, not being taken into account in any other way.

But, this might be bike-shedding (sorry for that), and I certainly don't have a _strong_ opinion either way. Both versions will work just fine.